src/Promise.php

```diff
@@ -97,6 +99,18 @@ public function cancel()
         $this->call($canceller);
     }
 
+    public function when(callable $onResolved)
+    {
+        $this->done(function ($value) use ($onResolved) {
+            $onResolved(null, $value);
+        }, function ($reason) use ($onResolved) {
+            $onResolved(
+                UnhandledRejectionException::nullOrResolve($reason),
+                null
+            );
+        });
```

Comment:
Yes, but since we're requiring `^0.2.0` and do not require `async-interop/event-loop` there is no other way at the moment.